app/controllers/dashboard/maps_controller.rb

```diff
@@ -2,14 +2,18 @@
 
 module Dashboard
   class MapsController < DashboardController
-    include MapsControllable
-   
-    load_and_authorize_resource
+    def show
+      @map = Map.find(params[:id])
+      @bounds = @map.bounds
+      @markers_json = @map.markers.to_json
+      @location_base_url = "/dashboard/maps/#{@map.id}/locations/"
+      @ask_location_permission = false
```

Comment:
Thanks for the feedback dude! Glad you like it and very good points raised. Some follow up questions on two points:

> Changing the lat or long values doesn't move the pin in the map, so as an user I'm not sure if it worked or not and where the pin now is displayed.

Ah good to know that you used it this way. I actually had in mind that the user will drag the marker, and not add lat / long inputs manually. In that case, I should disable the lat / longitude fields. Do you see a use case where the user would prefer to enter lat / long values instead of dragging the marker?

>  Users might think that they can search their locations and are then disappointed when it doesn't work. Later, we can replace it with an actual search bar that helps users find their locations

Yeah good idea; mapbox geocoding can actually be superpowered to also show our locations as results (see [example](https://docs.mapbox.com/mapbox-gl-js/example/forward-geocode-custom-data/)). This could be a nice addition, it doesn't seem to complicated to add, I'll see if I can add it quickly. It will then also solve the "show a list of locations in my dashboard" problem.

Reply:
To be clear, I was referring to the  `@ask_location_permission = false` assignment.

Reply:
We could handle that outside of the controller, but then you end up with logic in the view (if action is x then do y, etc.), which I don't find great either. The reason why I have to add these variables here is because I've tried to dry up the view (share the same between dashboard & public) & I think when we need to set too many variables, it's a good sign to un-dry the view & slip it into two separate views

Reply:
Ok, up to you. Just saying that some logic in the view is ok, and there's a reason the `[current_page?](https://apidock.com/rails/ActionView/Helpers/UrlHelper/current_page%3f)` view helper exists. Often, just assining bool variables in controller to be used in view is also code smell. As I said, up to you. I can live with it :-)